### PR TITLE
Feature/24490 Preserve preparedCheck entry in practice mode

### PR DESCRIPTION
**Pull request by GeorgeChatzigiannis**
- On pupil app practice config variable is passed to the check-started payload 
- `check-started` function can identify if the check is running on familiarisation and prevent deletion of preparedCheck

### CODE REVIEW

File: functions/check-started/index.js [hunk 1]
```diff
@@ -32,9 +32,11 @@ module.exports = async function (context, checkStartMessage) {
     throw error
   }
 
-  // Delete the row in the preparedCheck table - prevent pupils logging in again.
+  // Delete the row in the preparedCheck table for live checks only - prevent pupils logging in again.
   try {
```
Comment (GuyHarwood) [inline functions/check-started/index.js]: the problem with this is that you are trusting the client payload to be correct.  You should be looking this up in the main database via checkCode to determine how the flag is set.

File: functions/check-started/index.js [hunk 2]
```diff
@@ -32,10 +33,14 @@ module.exports = async function (context, checkStartMessage) {
     throw error
   }
 
-  // Delete the row in the preparedCheck table - prevent pupils logging in again.
+  // Delete the row in the preparedCheck table for live checks only - prevent pupils logging in again.
   try {
-    await deleteFromPreparedCheckTableStorage(azureTableService, checkStartMessage.checkCode, context.log)
-    context.log('SUCCESS: pupil check row deleted from preparedCheck table')
+    const check = await sqlFindCheckByCheckCode(checkStartMessage.checkCode, context.log)
+    context.log('check: ', check)
```
Comment (jon-shipley) [inline functions/check-started/index.js]: This check var looks like debug. Perhaps it should be removed?

File: functions/check-started/index.js [hunk 3]
```diff
@@ -32,10 +33,14 @@ module.exports = async function (context, checkStartMessage) {
...
 
-  // Delete the row in the preparedCheck table - prevent pupils logging in again.
+  // Delete the row in the preparedCheck table for live checks only - prevent pupils logging in again.
   try {
-    await deleteFromPreparedCheckTableStorage(azureTableService, checkStartMessage.checkCode, context.log)
-    context.log('SUCCESS: pupil check row deleted from preparedCheck table')
+    const check = await sqlFindCheckByCheckCode(checkStartMessage.checkCode, context.log)
+    context.log('check: ', check)
+    if (check.isLiveCheck) {
+      await deleteFromPreparedCheckTableStorage(azureTableService, checkStartMessage.checkCode, context.log)
```
Comment (jon-shipley) [inline functions/check-started/index.js]: 💯

File: functions/check-started/index.js [hunk 4]
```diff
@@ -91,3 +97,26 @@ async function updateAdminDatabaseForCheckStarted (checkCode, startedAt, logger)
...
   }
 }
+
+/**
+ * Fetch check record based on check code
+ * @param {String} checkCode - the unique GUID that identifies the check in the admin DB
+ * @param {Function} logger
+ * @return {Object} check record
+ */
+async function sqlFindCheckByCheckCode (checkCode, logger) {
```
Comment (jon-shipley) [inline functions/check-started/index.js]: You don't need to write this function again - just import '../lib/sql-helper' into the function.  There is an example usage like this in completed-check.

File: functions/check-started/index.js (hunk 2 above)
Comment (GuyHarwood) [reply functions/check-started/index.js]: i would agree.  if the object is sizeable then it creates a lot of noise in the logs

Comment (GeorgeChatzigiannis) [reply functions/check-started/index.js]: @jon-shipley @GuyHarwood this is removed.

File: functions/check-started/index.js (hunk 4 above)
Comment (GeorgeChatzigiannis) [reply functions/check-started/index.js]: thanks @jon-shipley, didn't spot that helper. I have updated based on that recommendation.

**Pull request merged**